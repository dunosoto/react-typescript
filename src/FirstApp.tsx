interface FirstAppProps {
  name: string,
  subTitle: string
}

export const FirstApp = ({name, subTitle}: FirstAppProps): JSX.Element => {
  
  const newMessage: string = 'Daniel';

  return (
    // jsx does not render Objects..
    <>
      <h1 data-testid='test-name'>{name}</h1>
      <p>{subTitle}</p>
    </>
  )
}
