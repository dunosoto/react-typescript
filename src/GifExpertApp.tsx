import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState<string[]>(['one punch']);

  const addNewCategory = (newCategory: string): void => {
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onAddNewCategory={ addNewCategory }/>

      {
        categories.map((category: string) => (
          <GifGrid category={category} key={category} />
        ))
      }
    </>
  )
}
