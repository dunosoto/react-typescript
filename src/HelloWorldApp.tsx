import PropTypes from 'prop-types';

export const HelloWorldApp = ():JSX.Element => {
  return (
    <>
      <h1>HelloWorldApp</h1>
    </>
  )
}

//proptypes for define types props. this is not necessary for typescript
HelloWorldApp.propTypes = {
  title: PropTypes.string.isRequired
}

HelloWorldApp.defaultPropTypes = {
  title: 'this a title'
}