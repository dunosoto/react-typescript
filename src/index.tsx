import React from 'react';
import ReactDom from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './components/CounterApp';
import './style.css';

ReactDom.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <CounterApp value={100} />
    {/* <FirstApp name='Daniel' subTitle='uno'/> */}
  </React.StrictMode>
);