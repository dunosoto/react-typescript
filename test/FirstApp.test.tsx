import React from 'react';
import { render, RenderResult, screen } from "@testing-library/react";
import {FirstApp} from "../src/FirstApp";

describe(' test in <FirstApp />', () => {

  const name: string =  'Daniel';

  test('should to do match with the snapshot', () => {
    const {container}: RenderResult = render(<FirstApp name={name} subTitle="subtitle"/>);
    
    expect(container).toMatchSnapshot();
  });

  test('should display the name into h1 tag', () => {

    const name: string =  'Daniel';
    const {container, getByText, getByTestId}: RenderResult = render(<FirstApp name={name} subTitle="subtitle"/>);

    
    expect(container).toMatchSnapshot();
    expect(getByText(name)).toBeTruthy();
    expect(getByTestId('test-name').innerHTML).toBe(name);
  });

  test('should display the subtitle into p tag sent by props', () => {

    const name: string =  'Daniel';
    const subtitle: string =  'I am a subtitle';
    const {getByText}: RenderResult = render(<FirstApp name={name} subTitle={subtitle} />);

    
    expect(getByText(subtitle)).toBeTruthy();
  });
});