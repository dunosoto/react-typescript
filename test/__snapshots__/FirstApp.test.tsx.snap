// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test in <FirstApp /> should display the name into h1 tag 1`] = `
<div>
  <h1
    data-testid="test-name"
  >
    Daniel
  </h1>
  <p>
    subtitle
  </p>
</div>
`;

exports[` test in <FirstApp /> should to do match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-name"
  >
    Daniel
  </h1>
  <p>
    subtitle
  </p>
</div>
`;
